src/fcc_bridge_node_commands.cpp

```diff
@@ -0,0 +1,465 @@
+    switch (this->get_internal_state()) {
+        case INTERNAL_STATE::ERROR:
+            // This should never happen as the process should have exited before
+            throw std::runtime_error(
+                "Received command to take of while in ERROR state");
+        case INTERNAL_STATE::STARTING_UP:
+        case INTERNAL_STATE::ROS_SET_UP:
+        case INTERNAL_STATE::MAVSDK_SET_UP:
+        case INTERNAL_STATE::WAITING_FOR_ARM:
+        case INTERNAL_STATE::LANDED:
+        case INTERNAL_STATE::ARMED:
+            // The UAV is still or again on the ground so the only thing left is
+            // to kill the process
+            RCLCPP_FATAL(
+                this->get_logger(),
+                "Received a takeoff command in an invalid state! Exiting...");
+            this->set_internal_state(INTERNAL_STATE::ERROR);
+            this->exit_process_on_error();
+        case INTERNAL_STATE::FLYING_MISSION:
+        case INTERNAL_STATE::LANDING:
+        case INTERNAL_STATE::RETURN_TO_HOME:
+            // If the UAV is already waiting on a UAVCommand and another
+            // command is received something must have gone wrong so an RTH is
+            // triggered
+            RCLCPP_ERROR(this->get_logger(),
+                         "Received a waypoint command while there is already a "
+                         "command in progress! Triggering RTH...");
+            this->trigger_rth();
+            return;
+        case INTERNAL_STATE::WAITING_FOR_COMMAND:
+            break;
+        default:
+            throw std::runtime_error(
+                std::string("Got invalid value for internal_state: ") +
+                std::to_string(static_cast<int>(this->get_internal_state())));
+    }
+
+    // Check that the speed is valid
+    if (!this->check_speed(speed_mps)) {
+        if (this->get_internal_state() == INTERNAL_STATE::ERROR) {
+            // This means an unrecoverable error has occurred such as missing
+            // safety limits
+            RCLCPP_FATAL(
+                this->get_logger(),
+                "There has been an unrecoverable error checking if the target "
+                "speed for waypoint was inside the safety limits! Exiting...");
+            // Exit the process
+            this->exit_process_on_error();
+        } else {
+            // In this case the target waypoint speed is outside the limits
+            // deemed safe. This will result in an RTH
+            RCLCPP_ERROR(
+                this->get_logger(),
+                "The target waypoint speed is outside the speed safety limits "
+                "(%f is not in (%f;%f]). Triggering RTH...",
+                static_cast<double>(speed_mps),
+                static_cast<double>(safety_limits::MIN_SPEED_LIMIT_MPS),
+                static_cast<double>(this->safety_limits->max_speed_mps));
+            this->trigger_rth();
+            return;
+        }
+    }
+
+    // Check that the waypoint is valid
```

Comment:
Maybe write a helper function for this as it seems to be the same as in lines 88-96?

Reply:
See https://github.com/THI-Drone/fcc_bridge_package/pull/1/files/4f59a0a68753e2e1f18bd92ee57894f2e2e4387d?diff=unified&w=0#r1597623132